File: LeetCode/java/950_Reveal_Cards_In_Increasing_Order.java
import java.util.ArrayDeque;
import java.util.Arrays;
import java.util.Deque;

class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        int[] answer = new int[deck.length];
        Deque<Integer> dq = new ArrayDeque<>();

        Arrays.sort(deck);

        for (int i = deck.length - 1; i >= 0; i--) {
            if (!dq.isEmpty()) {
                int n = dq.pollLast();
                dq.offerFirst(n);
            }

            dq.offerFirst(deck[i]);
        }

        int idx = 0;
        while (!dq.isEmpty()) {
            answer[idx++] = dq.pollFirst();
        }

        return answer;
    }
}

File: LeetCode/cpp/950_Reveal_Cards_In_Increasing_Order.cpp
class Solution {
public:
	vector<int> deckRevealedIncreasing(vector<int>& deck) {
		vector<int> answer;
		deque<int> dq;

		sort(deck.begin(), deck.end());

		for (int i = deck.size() - 1; i >= 0; i--) {
			if (!dq.empty()) {
				int n = dq.back();
				dq.pop_back();
				dq.push_front(n);
			}

			dq.push_front(deck[i]);
		}

		while (!dq.empty()) {
			answer.push_back(dq.front());
			dq.pop_front();
		}

		return answer;
	}
};